test/nodes/test_retriever.py

```diff
@@ -650,6 +650,37 @@ def test_table_text_retriever_training(tmp_path, document_store, samples_path):
     )
 
 
+def test_sentence_transformers_retriever_training_with_gradient_checkpointing():
+    def mock_fit(*args, **kwargs):
+        return None
+
+    with patch("sentence_transformers.SentenceTransformer.fit", mock_fit):
+        retriever = EmbeddingRetriever(
+            embedding_model="sentence-transformers/all-MiniLM-L6-v2",
+            model_format="sentence_transformers",
+            use_gpu=False,
+        )
+        retriever.train(
+            training_data=[
+                {
+                    "question": "What color is the sky?",
+                    "pos_doc": "The color of the sky is blue.",
+                    "neg_doc": "The color of the grass is brown.",
+                },
+                {
+                    "question": "What is the capital of Germany?",
+                    "pos_doc": "The capital of Germany is Berlin.",
+                    "neg_doc": "The capital of France is Paris.",
+                },
+            ],
+            train_loss="mnrl",
+            n_epochs=1,
+            batch_size=1,
+            num_warmup_steps=0,
+            gradient_checkpointing=True,
+        )
```

Comment:
Yes that's the idea! What you can do now is to make sure that `mock_fit` is called (once I guess?) with the right parameters, like:

```suggestion
def test_sentence_transformers_retriever_training_with_gradient_checkpointing():
    with patch("sentence_transformers.SentenceTransformer.fit") as mock_fit:
        retriever = EmbeddingRetriever(
            embedding_model="sentence-transformers/all-MiniLM-L6-v2",
            model_format="sentence_transformers",
            use_gpu=False,
        )
        retriever.train(
            training_data=[
                {
                    "question": "What color is the sky?",
                    "pos_doc": "The color of the sky is blue.",
                    "neg_doc": "The color of the grass is brown.",
                },
                {
                    "question": "What is the capital of Germany?",
                    "pos_doc": "The capital of Germany is Berlin.",
                    "neg_doc": "The capital of France is Paris.",
                },
            ],
            train_loss="mnrl",
            n_epochs=1,
            batch_size=1,
            num_warmup_steps=0,
            gradient_checkpointing=True,
        )
        mock_fit.assert_called_once_with(... the parameters...)
```

Just fill in the parameters the function is called with and the PR is ready to go :slightly_smiling_face: